### PR TITLE
Add button to refresh session id

### PR DESCRIPTION
**Pull request by ThiloAschebrock**
image

### CODE REVIEW

File: frontend/src/Components/LoginPage/LoginPage.module.css [hunk 1]
```diff
@@ -89,3 +83,41 @@
...
+.sessionLink {
+  color: var(--primary);
+  min-width: 19ch;
+}
+
+.refreshButton {
+  color: var(--primary);
+  fill: var(--primary);
+  border: unset;
+  border: 1px solid var(--primary);
```
Comment (lukastaegert) [inline frontend/src/Components/LoginPage/LoginPage.module.css:105]: Wouldn't the second border value just override the first value, making the first value unnecessary?

File: frontend/src/Components/LoginPage/LoginPage.module.css [hunk 2]
```diff
@@ -89,3 +83,41 @@
...
+  background: var(--secondary);
+  border-radius: var(--border-radius);
+  cursor: pointer;
+  height: 22px;
+  width: 22px;
+  padding: 2px;
+  display: block;
+  line-height: unset;
```
Comment (lukastaegert) [inline frontend/src/Components/LoginPage/LoginPage.module.css:113]: That feels dangerous to me, but maybe I am wrong. The parent does not really have an explicit line-height, so it is set to "normal", which is again something browser dependent. Considering that having a line-height that is too much would look bad, why not set it explicitly e.g. to 1, which appears to look about right?

File: frontend/src/Components/LoginPage/LoginPage.tsx [hunk 3]
```diff
@@ -65,9 +59,23 @@ export const LoginPage = connectToWebSocket(({ socket }) => {
...
+          {sessionId}
+        </a>
+        <button
+          type="button"
+          class={classes.refreshButton}
+          onClick={() => setSessionId(refreshSessionId())}
+          title={BUTTON_REFRESH_SESSION}
+        >
+          {/* icon taken from material ui icons */}
+          <svg xmlns="http://www.w3.org/2000/svg" height="16px" viewBox="0 0 24 24" width="16px">
```
Comment (lukastaegert) [inline frontend/src/Components/LoginPage/LoginPage.tsx:73]: Could we either extract the icon or the entire button (similar to the "KickButton") to a separate component for readability?

**Review by lukastaegert — COMMENTED** [review]
Looks good to me, just some small comments. One thing I noticed that felt slightly off to me was, that when I clicked the button, it stayed in its "blue" state and no longer had any changes on hover. That is of course because hover and focus have the same design and is consistent with the kick button. For the kick button, however, there was not really an issue because the button vanished after clicking. I wonder if we want to have e.g. a slight change in shade on hover when it is focused?

File: frontend/src/Components/LoginPage/LoginPage.module.css (hunk 2 above)
Comment (ThiloAschebrock) [reply frontend/src/Components/LoginPage/LoginPage.module.css:113]: Going with the height of the icon, i.e., 16px.

File: frontend/src/Components/LoginPage/LoginPage.tsx (hunk 3 above)
Comment (ThiloAschebrock) [reply frontend/src/Components/LoginPage/LoginPage.tsx:73]: Extracted the button.

**Comment by ThiloAschebrock** [comment]
I change the focus state to use outline similar to our links instead to separate from the hover state as suggested.

**Review by lukastaegert — APPROVED** [review]
Thanks, looks good to me!

**Pull request merged**